phpBB/includes/functions_download.php

```diff
@@ -69,24 +63,23 @@ function send_avatar_to_browser($file, $browser)
 			header('Expires: ' . gmdate('D, d M Y H:i:s', time() + 31536000) . ' GMT');
 		}
 
-		$size = @filesize($file_path);
-		if ($size)
-		{
-			header("Content-Length: $size");
+		try {
+			header('Content-Type: ' . $file_info->size);
+		} catch (\phpbb\storage\exception\not_implemented $e) {
+			// Just dont send this header
 		}
 
-		if (@readfile($file_path) == false)
-		{
-			$fp = @fopen($file_path, 'rb');
+		try {
+			$fp = $storage->read_stream($file_path);
 
-			if ($fp !== false)
+			while (!feof($fp))
 			{
-				while (!feof($fp))
-				{
-					echo fread($fp, 8192);
-				}
-				fclose($fp);
+				echo fread($fp, 8192);
```

Comment:
actually as we have a stream we can use `stream_copy_to_stream($fp, 'php://output')` http://php.net/manual/fr/function.stream-copy-to-stream.php